tesseract_python/tesseract_python/CMakeLists.txt

```diff
@@ -0,0 +1,191 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(tesseract_python)
+
+# Read version
+FILE (STRINGS "${CMAKE_CURRENT_SOURCE_DIR}/package.xml" tesseract_python_version1 REGEX "<version>[0-9]+\\.[0-9]+\\.[0-9]+</version>")
+IF ("${tesseract_python_version1}" STREQUAL "" )
+MESSAGE(FATAL_ERROR "Could not read tesseract_python version.")
+ENDIF()
+STRING (REGEX REPLACE "<version>([0-9]+\\.[0-9]+\\.[0-9]+)</version>" "\\1" tesseract_python_version ${tesseract_python_version1})
+MESSAGE(STATUS "tesseract_python version: ${tesseract_python_version}")
```

Comment:
This looks useful so you only have to update the version in the package.xml. What are your thoughts on using this for every package? I am not asking to make the change now, but if you agree I will create a PR after this is merged adding this to tesseract_common cmake macro file with a function called extract_version.

Reply:
Where do you need version information in the rest of Tesseract? The version information here is mainly for use with building Python wheels. For the ROS packages you probably only need the package.xml version.

I based this version extraction of this code in Robot Raconteur: https://github.com/robotraconteur/robotraconteur/blob/2ae3738e937e72c7b8096914d29e3bbeb379b71f/CMakeLists.txt#L10-L17

On a related note, it would be useful to be able to build the Python module with static dependencies so there is only the one *.so or *.pyd binary file in the Python wheel. It would make distribution a lot easier. That can be done after the pull request is accepted.

Reply:
It would actually not be a bad idea to set the `VERSION` argument to `project(..)`, as it would then generate version-specific artefacts. If we forward the version to libraries as well, that would help prevent ABI compatibility problems because CMake would set `soversions` if we ask it to (which helps explicitly error out when users don't rebuild their workspace properly after updating packages).

In another project I've used this set of statements, which is a bit more concise:

```cmake
# Read version from the package.xml file.
file(READ ${CMAKE_CURRENT_SOURCE_DIR}/package.xml package_xml_str)
if(NOT package_xml_str MATCHES "<version>([0-9]+.[0-9]+.[0-9]+)</version>")
  message(FATAL_ERROR "Could not parse project version from package.xml. Aborting.")
endif()

# At this point we either have a proper version string, or we've errored
# out with a FATAL_ERROR above. So assume CMAKE_MATCH_1 contains our
# package's version.
project(... VERSION ${CMAKE_MATCH_1} ...)
```

All plain CMake packages get their versions from the package manifest in this way. Then use `set_target_properties(... VERSION ${${PROJECT_NAME}_VERSION ...)` on libraries and other targets where it makes sense.

Reply:
@gavanderhoorn I will added to all of the packages! :)